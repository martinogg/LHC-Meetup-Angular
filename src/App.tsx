/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 * 
 * Generated with the TypeScript template
 * https://github.com/emin93/react-native-template-typescript
 * 
 * @format
 */

import { createAppContainer, createStackNavigator } from 'react-navigation'
import React, { Component } from 'react';
import { createStore, combineReducers, applyMiddleware } from 'redux'
import { Provider } from 'react-redux'
import logger from 'redux-logger'

import { AboutAppScreen } from './Screens/AboutApp/AboutAppScreen'
import { ProfileScreen } from './Screens/Profile/ProfileScreen'
import { HomeScreen } from './Screens/Home/HomeScreen'
import { LoginScreen } from './Screens/Login/LoginScreen'
import { RegisterScreen } from './Screens/Register/RegisterScreen'
import { BrowseScreen } from './Screens/Browse/BrowseScreen'
import { InterestScreen } from './Screens/Interest/InterestScreen'
import FirebaseConnection from './Helpers/FirebaseConnection'

import AReducer from './Redux/Reducers/DummyReducer'

const MainNavigator = createStackNavigator({
  Login: { screen: LoginScreen },
  Register: { screen: RegisterScreen },
  Profile: { screen: ProfileScreen },
  Home: { screen: HomeScreen },
  AboutApp: { screen: AboutAppScreen },
  Browse: { screen: BrowseScreen },
  Interest: { screen: InterestScreen }
},
  { initialRouteName: 'Login' });

const Nav = createAppContainer(MainNavigator);

const reducer = combineReducers({ AReducer })
const store = createStore(reducer, applyMiddleware(logger))

const scrProps = { firebaseConnection: FirebaseConnection.getInstance() }

interface Props { }
class App extends Component<Props> {

  public render() {
    return (
      <Provider store={store}>
        <Nav screenProps={scrProps} />
      </Provider>
    )
  }
}

export default App;
