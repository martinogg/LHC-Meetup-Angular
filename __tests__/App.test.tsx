/**
 * @format
 */

 // TODO -- Tidy this up

import React from 'react';
import 'react-native';

import { shallow } from 'enzyme'

import App from '../src/App'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

/*
// Add 'export' to fake this being a module to silence TSLint.
export const add = (a: number, b: number) => a + b;
describe("add", () => {
  it("should add two numbers", () => {
    expect(add(1, 1)).toEqual(2);
  });
});
*/

export const createTestProps = (props: object) => ({
  ...props
});

it('should work', () => {

  expect(1).toBe(1)
})

/* // TODO : Test App Class (navigation)

describe("App", () => {
  const props = createTestProps({});
  const wrapper = shallow(<App {...props} />);
                               
  describe("rendering", () => {
    it("should render a <View />", () => {
      expect(wrapper.find("View")).toHaveLength(1);
    });
  });
});
*/