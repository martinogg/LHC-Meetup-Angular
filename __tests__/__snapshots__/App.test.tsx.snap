// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display HomeScreen with no errors 1`] = `
<View
  accessibilityRole="button"
  accessibilityStates={Array []}
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {},
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#007AFF",
            "fontSize": 18,
            "padding": 8,
            "textAlign": "center",
          },
        ]
      }
    >
      Go to Jane's profile
    </Text>
  </View>
</View>
`;

exports[`should display ProfileScreen with no errors 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 20,
        "margin": 10,
        "textAlign": "center",
      }
    }
  >
    Welcome to React native!
  </Text>
  <Text
    style={
      Object {
        "color": "#333333",
        "marginBottom": 5,
        "textAlign": "center",
      }
    }
  >
    To get started, edit App.tsx
  </Text>
  <Text
    style={
      Object {
        "color": "#333333",
        "marginBottom": 5,
        "textAlign": "center",
      }
    }
  >
    Press Cmd+R to reload,
Cmd+D or shake for dev menu
  </Text>
</View>
`;
